packages/rest/src/validation/request-body.validator.ts

```diff
@@ -87,10 +87,17 @@ function convertToJsonSchema(openapiSchema: SchemaObject) {
  * @param schema The JSON schema used to perform the validation.
  * @param globalSchemas Schema references.
  */
+export interface AnyObject {
+  // tslint:disable-next-line:no-any
+  [property: string]: any;
+}
+
+const compiledSchemaCache = new WeakMap();
```

Comment:
Singletons are usually an anti-pattern to avoid.

How is this cache going to behave in regards to memory management and garbage collection? For example, when a route (API endpoint) or perhaps an entire LB4 app is deleted (removed from the process), when will the cached validators be released by V8?

Have you considered storing the compiled validators in Route objects as have been proposed in the issue description?

Let's discuss!

Reply:
WeakMap keys are just references to object IDs, so we don't have to worry about memory management and garbage collection.

I opted for a caching object right in the validator for the sake of simplicity. Would this cache be ever accessed from outside the validator? If not, then I am not very convinced it should be located outside the validator.

Reply:
> WeakMap keys are just references to object IDs, so we don't have to worry about memory management and garbage collection.

How about the values stored in the map? When are those garbage collected?

I'd like to avoid the situation when removing a controller from the app does not release all associated memory. This used to be a problem in LoopBack 2.x/3.x and it was non-trivial to discover all sources of leaks.